semanticdb/semanticdb/semanticdb.proto

```diff
@@ -278,7 +278,7 @@ message SymbolInformation {
   int32 properties = 4;
   string display_name = 5;
   Signature signature = 17;
-  repeated Annotation annotations = 13;
+  repeated AnnotationTree annotations = 13;
```

Comment:
```suggestion
  repeated AnnotationTree annotations = 13; // NOTE: it's possible to manually change the type to `Annotation` for source compatibility with older SemanticDB versions.
```

Reply:
@tgodzik @gabro I'm curious to hear your thoughts about this Protobuf binary-compatible but Scala binary+source-incompatible change. It's possible to keep the old `Annotation` signature to preserve source compatibility but the new `AnnotationTree` is more consistently named with other `Tree` subclasses. 

A few sidenotes:

- The ScalaPB-generated classes have intentionally never been marked as public APIs exactly so that we can make Protobuf binary-compatible changes like this.
- SemanticDB annotations are not really used by any meaningful Scalameta project from what I can tell because they are incomplete for Scala (no parameters until this PR) and they were officially unsupported for Java.
- Scalafix will be able to incorporate these changes into its public API without breaking changes.

WDYT?